### ISSUE TITLE
"Discard modified lines..." is broken

### ISSUE THREAD

**Issue by aybe** [desktop/desktop#19426]
### The problem

1. I first try without hiding whitespace changes, it works
2. Next, I enable with hiding whitespace changes enabled, it does nothing and the wizard to turn it off doesn't appear anymore
3. Finally, when opening the options, and if trying to click something else, the action isn't executed and the settings stay open.

https://github.com/user-attachments/assets/3fb873c8-cfa4-4ee3-9a3c-af03652ad246

Also, notice how at some point, 44 minutes at the bottom becomes 1 hour ago for no reason...

### Release version

Version 3.4.7-beta2 (x64)

### Operating system

Windows 11

**Comment by sergiou87** [comment]
The behavior (not being able to discard those files) is correct: when whitespace changes are hidden, we can't tell which lines in the original diff are you pointing at, hence why we disable it.

However, we should give some feedback about that. For example, if you click with your left mouse button you get this:

![Image](https://github.com/user-attachments/assets/ad300378-6955-469d-90ac-516465a751a6)

**Comment by DylanDevelops** [comment]
I could try working on this if you would like!

**Comment by aybe** [comment]
@sergiou87
Yes, you are right... Go figure why I was pressing the other mouse button...
An improvement could be that the message box also appears when pressing the right mouse button.

**Comment by Morinyo1907** [comment]
> ### The problem
> 1. I first try without hiding whitespace changes, it works
> 2. Next, I enable with hiding whitespace changes enabled, it does nothing and the wizard to turn it off doesn't appear anymore
> 3. Finally, when opening the options, and if trying to click something else, the action isn't executed and the settings stay open.
> 
>  bandicam.2024-10-21.02-08-32-205.mp4 
> Also, notice how at some point, 44 minutes at the bottom becomes 1 hour ago for no reason...
> 
> ### Release version
> Version 3.4.7-beta2 (x64)
> 
> ### Operating system
> Windows 11
> 
> ### Steps to reproduce the behavior
> _No response_
> 
> ### Log files
> _No response_
> 
> ### Screenshots
> _No response_
> 
> ### Additional context
> _No response_

0x7Bb8B7Eb88EDCd255427094e58ADD138BA52B177
